sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala

```diff
@@ -1570,7 +1570,7 @@ class Dataset[T] private[sql](
    * @since 2.0.0
    */
   @scala.annotation.varargs
-  def selectExpr(exprs: String*): DataFrame = sparkSession.withActive {
```

Comment:
TBH I am not sure what is worse. The problem I foresee here is that expressions use SQLConf.get to get a configuration, and now we don't know what setting they will pick-up.

Reply:
That's a good point. It's not only about the parser, but also expression instantiations.

@yaooqinn can you fix other DF APIs to wrap with `sparkSession.withActive`? I believe there are more APIs that take a SQL string and parse it.

Reply:
`spark.sql("select '2023-01-01' as dt").select(expr("dt + INTERVAL 1 YEAR as dt1"))`, `expr`  use the active SQLConf, should it be modified?

Reply:
`expr` is a static function. It can only use the active sql conf.

Reply:
I understand, thanks for clarifying.